search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -309,6 +318,130 @@ class AttributeInstanceService(
         )
     }
 
+    private suspend fun retrieveTeaUUIDAndPayload(
+        entityId: URI,
+        attributeName: ExpandedTerm,
+        instanceId: URI
+    ): Either<APIException, Pair<UUID, Map<String, List<Any>>>> {
+        val modifiedQuery =
+            """
+            SELECT temporal_entity_attribute, payload
+            FROM attribute_instance
+            WHERE temporal_entity_attribute = ( 
+                SELECT id 
+                FROM temporal_entity_attribute 
+                WHERE entity_id = :entity_id 
+                AND attribute_name = :attribute_name
+            )
+            AND instance_id = :instance_id
+            """.trimIndent()
+
+        return databaseClient
+            .sql(modifiedQuery)
+            .bind("entity_id", entityId)
+            .bind("attribute_name", attributeName)
+            .bind("instance_id", instanceId)
+            .oneToResult(
+                ResourceNotFoundException(
+                    attributeOrInstanceNotFoundMessage(attributeName, instanceId.toString())
+                )
+            ) {
+                Pair(
+                    it["temporal_entity_attribute"] as UUID,
+                    JsonUtils.deserializeObject(toJsonString(it["payload"])) as Map<String, List<Any>>
+                )
+            }
+    }
+
+    @Transactional
+    suspend fun modifyAttributeInstance(
+        entityId: URI,
+        attributeName: ExpandedTerm,
+        instanceId: URI,
+        expandedPayload: Map<String, List<Any>>
+    ): Either<APIException, Unit> = either {
+        val (teaUUID, oldPayload) = retrieveTeaUUIDAndPayload(entityId, attributeName, instanceId).bind()
+        val payload = expandedPayload.toMutableMap()
+        oldPayload.forEach {
+            payload.putIfAbsent(it.key, it.value)
+        }
```

Comment:
it should be a replacement, not a merge ("Replace the target Attribute instance identified by the instanceId with the Attribute instance in the EntityTemporal Fragment" - 5.6.14.4)

Reply:
Here I am using the old payload to build the new one. I only add the elements that already exist in the old payload that are not in the new one. For example, if the user in the fragment does not put the type it is automatically added using the old payload. We also have the instance id in the payload which is added to the new payload using the old one. (